chsdi/lib/opentransapi/opentransapi.py

```diff
@@ -0,0 +1,95 @@
+            'estimatedDate': self._convert_estimated_date(el.find('./trias:ThisCall/trias:CallAtStop/trias:ServiceDeparture/trias:EstimatedTime', ns)),
+            'destinationName': el.find('./trias:Service/trias:DestinationText/trias:Text', ns).text,
+            'destinationId': el.find('./trias:Service/trias:DestinationStopPointRef', ns).text
+        } for el in el_stop_events]
+        if el_stop_events == None or results == []:
+            results = [{'destination': 'nodata'}]
+        return results
+
+    def send_post(self, station_id, request_dt_time, number_results=5):
+        self.headers = {
+            'authorization': self.open_trans_api_key,
+            'content-type': 'application/xml; charset=utf-8',
+            'accept-charset': 'utf-8'
+        }
+        self.xml_data = u"""
+                    <?xml version="1.0" encoding="UTF-8"?>
+                    <Trias version="1.1" xmlns="http://www.vdv.de/trias" xmlns:siri="http://www.siri.org.uk/siri" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+                        <ServiceRequest>
+                            <siri:RequestTimestamp>%s</siri:RequestTimestamp>
+                            <siri:RequestorRef>EPSa</siri:RequestorRef>
+                            <RequestPayload>
+                                <StopEventRequest>
+                                    <Location>
+                                        <LocationRef>
+                                            <StopPointRef>%s</StopPointRef>
+                                        </LocationRef>
+                                        <DepArrTime>%s</DepArrTime>
+                                    </Location>
+                                    <Params>
+                                        <NumberOfResults>%s</NumberOfResults>
+                                        <StopEventType>departure</StopEventType>
+                                        <IncludePreviousCalls>false</IncludePreviousCalls>
+                                        <IncludeOnwardCalls>true</IncludeOnwardCalls>
+                                        <IncludeRealtimeData>true</IncludeRealtimeData>
+                                    </Params>
+                                </StopEventRequest>
+                            </RequestPayload>
+                        </ServiceRequest>
+                    </Trias>
+                    """ % (str(request_dt_time), str(station_id), str(request_dt_time), str(number_results))
+
+        self.r = requests.post(self.url, self.xml_data, headers=self.headers, timeout=5)
+        if (self.r.status_code == 429):
+            raise OpenTransException("The rate limit of OpenTransportdata has exceeded")
+        if (self.r.status_code != requests.codes.ok):
+            self.r.raise_for_status()
+
+        self.r.encoding = 'utf-8'  # TODO better encoding solution
+        return self.r.text.encode('utf-8')
```

Comment:
what is in `r.encoding` ?

Reply:
http://docs.python-requests.org/en/master/user/quickstart/

-> set the encoding manually